rmgpy/molecule/group.py

```diff
@@ -1021,6 +1021,33 @@ def update(self):
         self.updateConnectivityValues()
         self.updateFingerprint()
 
+    def updateCharge(self):
+        """
+        Update the charge according to the valence_electron, total bond order, lone pairs
+        and radical electrons. When charged molecule is appearing in reactants, the product
+        template generated by applying recipe will result in wrong charge status. This
+        function is aimed to solve this problem
+        """
+        for atom in self.atoms:
+            if (len(atom.charge) == 1) & (len(atom.lonePairs) == 1) & (len(atom.radicalElectrons) == 1):
+                # if the charge of the group is not labeled, then no charge update will be
+                # performed. If there is a list of charge, no update either. Besides, this groupatom have to
+                # have enough information to be updated
+                atomtype = atom.atomType[0]
+                for element in allElements:
+                    if atomtype is atomTypes[element] or atomtype in atomTypes[element].specific:
+                        elementlabel = element
+                        bondOrder, lonePairs, radicalElectrons = 0, 0, 0
+                        import rmgpy.molecule.element as elements
```

Comment:
I think that imports should generally be done either at the head of the file, or the start of the method.
Although it doesn't seem like the case here, an import in a for loop could get called several times needlessly.

Reply:
import has been moved to the head of the file